app/Helpers/Bank/Nordigen/Transformer/AccountTransformer.php

```diff
@@ -0,0 +1,121 @@
+            [usage] => PRIV
+        )
+    [metadata] => stdClass Object
+        (
+            [id] => XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+            [created] => 2022-12-05T18:41:53.986028Z
+            [last_accessed] => 2023-10-29T08:35:34.003611Z
+            [iban] => DE0286055592XXXXXXXXXX
+            [institution_id] => STADT_KREISSPARKASSE_LEIPZIG_WELADE8LXXX
+            [status] => READY
+            [owner_name] => Max Mustermann
+        )
+    [balances] => [
+                {
+                    [balanceAmount]: {
+                        [amount] => 9825.64
+                        [currency] => EUR
+                    },
+                    [balanceType] => closingBooked
+                    [referenceDate] => 2023-12-01
+                },
+                {
+                    [balanceAmount[: {
+                        [amount] => 10325.64
+                        [currency] => EUR
+                    },
+                    [balanceType] => interimAvailable
+                    [creditLimitIncluded]: true,
+                    [referenceDate] => 2023-12-01
+                }
+            ]
+    [institution] => stdClass Object
+        (
+            [id] => STADT_KREISSPARKASSE_LEIPZIG_WELADE8LXXX
+            [name] => Stadt- und Kreissparkasse Leipzig
+            [bic] => WELADE8LXXX
+            [transaction_total_days] => 360
+            [countries] => [
+                "DE"
+            ],
+            [logo] => https://storage.googleapis.com/gc-prd-institution_icons-production/DE/PNG/sparkasse.png
+            [supported_payments] => {
+                [single-payment] => [
+                    "SCT",
+                    "ISCT"
+                ]
+            },
+            [supported_features] => [
+                "card_accounts",
+                "payments",
+                "pending_transactions"
+            ],
+            [identification_codes] => []
+        )
+
+    )
+ */
+
+
+class AccountTransformer implements AccountTransformerInterface
+{
+
+    public function transform($nordigen_account)
+    {
+
+        if (!property_exists($nordigen_account, 'data') || !property_exists($nordigen_account, 'metadata') || !property_exists($nordigen_account, 'balances') || !property_exists($nordigen_account, 'institution'))
+            throw new \Exception('invalid dataset');
+
+        $used_balance = $nordigen_account->balances[0];
+        // prefer entry with closingBooked
+        foreach ($nordigen_account->balances as $entry) {
+            if ($entry["balanceType"] === 'closingBooked') { // available: closingBooked, interimAvailable
+                $used_balance = $entry;
+                break;
+            }
+        }
+
+        return [
+            'id' => $nordigen_account->metadata["id"], // TODO: maybe add prefix for unique id between yodlee and nordigen?
+            'account_type' => "bank_account", // TODO: not creditCard, which type should be used here?!
```

Comment:
Does Nordigen only return bank_account? if so that is fine, we'll just need to document for the front ends

Reply:
yes, they do.
are there some already used keywords, to set by default from me or should it may be empty?

Reply:
@paulwer if bank_account is the only keyword used, then this is fine to leave as bank_account.

Reply:
great